compiler/modules/types/core.bal

```diff
@@ -757,6 +764,30 @@ public function uniformTypeCode(UniformTypeBitSet bitSet) returns UniformTypeCod
     return <UniformTypeCode>lib:numberOfTrailingZeros(bitSet);
 }
 
+public function comparable(Context cx, SemType t1, SemType t2) returns boolean {
+    ComparableMemo? memoized = cx.comparableMemo[[t1, t2]];
+    if memoized != () {
+        return memoized.comparable;
+    }
+    ComparableMemo memo = { semTypes: [t1, t2] };
+    cx.comparableMemo.add(memo);
+    boolean comparable = comparableInternal(cx, t1, t2);
+    memo.comparable = comparable;
+    return comparable;
+}
+
+function comparableInternal(Context cx, SemType t1, SemType t2) returns boolean {
+    SemType semType = diff(union(t1, t2), NIL);
+    if isSubtypeSimple(semType, SIMPLE_OR_STRING) {
+        int nOrderings = lib:bitCount(widenToUniformTypes(semType));
+        return nOrderings <= 1;
+    }
+    if isSubtypeSimple(semType, LIST) {
+        return comparable(cx, listMemberType(cx, t1, ()), listMemberType(cx, t2, ()));
```

Comment:
Does this deal with the case where t1 or t1 has type `()`? If so, we should have a test for that.

We also need a SUBSET comment for tuples.

Reply:
There is already tests covering this [order2-v.bal](https://github.com/ballerina-platform/nballerina/blob/6da311d71760793685621edea27e5ccb3cf4a98d/compiler/testSuite/11-nested/order2-v.bal#L9).

`comparable(NIL, INT_ARR) = true` because `listMemberType(NIL) = NEVER` and `comparable(NEVER, INT) = true`.

Maybe it's wrong for `listMemberType` to return `NEVER` in this case.